File: 卡码/代码随想录/1/08-贪心算法/Js/01-简单/LeetCode_0455.js
//455. 分发饼干
const findContentChildren = function(g, s) {
  g.sort((a,b)=>a-b);
  s.sort((a,b)=>a-b);
  let res=0,index=s.length-1;
  for(let i=g.length-1;i>=0;i--){
    if(index<0) break;
    if(s[index]>=g[i]){
      res++;
      index--;
    }
  }
  return res;
};

File: 卡码/代码随想录/1/08-贪心算法/Python/01-简单/LeetCode_0455.py
#455. 分发饼干
from typing import *

class Solution:
  def findContentChildren(self, g: List[int], s: List[int]) -> int:
    g.sort()
    s.sort()
    res:int=0
    index:int=len(s)-1
    for i in range(len(g)-1,-1,-1):
      if index<0:
        break
      if s[index]>=g[i]:
        res+=1
        index-=1
    return res